### DISCUSSION TITLE
How can I speed up JBrowse2 load time when the configuration has many tracks?

### DISCUSSION THREAD

**Discussion by crisamor** [GMOD/jbrowse-components#4170] [General] [answered]
I have installed JBrowse2 in my server and uploaded about 6,000 tracks, but the load time now is too slow, about 20 seconds. Looking for some performance improvement I found that one option is to use types.frozen for jbrowseModel.tracks (https://github.com/GMOD/jbrowse-components/pull/3468 ), but I don't know how to implement this, do I have to install or develop any plugin? I would be very grateful if anyone could provide any clue on that issue. Thanks in advance!

**Comment by cmdcolin** [comment]
@crisamor hi there. There is indeed slowness with large numbers of tracks at load time. That PR was promising but was halted as it had some complicated implications. I could try and revisit that PR and build a beta branch if you are interested in trying it out. I can't guarantee it could get merged in any timeliness but would.be happy to make that happen

**Comment by cmdcolin** [comment, marked as answer]
@crisamor  you can try using this frozen_tracks3 branch

To download it using the CLI you can use

    jbrowse create --branch frozen_tracks3 newfolder #get new folder
    jbrowse upgrade --branch frozen_tracks3 #or, update your existing instance



here is a live demo showing the difference in data loading time, on about 1700 tracks

new branch
http://s3.amazonaws.com/jbrowse.org/code/jb2/frozen_tracks3/index.html?config=%2Fjbrowse.org%2Fgenomes%2FGRCh38%2F1000genomes%2Fconfig_1000genomes.json&session=share-M3ynDk4EAp&password=OVZ8U

main branch
http://s3.amazonaws.com/jbrowse.org/code/jb2/main/index.html?config=%2Fjbrowse.org%2Fgenomes%2FGRCh38%2F1000genomes%2Fconfig_1000genomes.json&session=share-M3ynDk4EAp&password=OVZ8U

**Comment by crisamor** [comment]
Hi @cmdcolin, thank you so much for your quick response! It would be interesting to try out the frozen_tracks3 branch. However, I'm new with JBrowse2 and I don't have much idea on how to implement the types.frozen for jbrowseModel.tracks option. Do I have to set up this option in the config.json file? I tried to follow the link to the 1000 genomes config file that you provided, but it doesn't work.
http://localhost:3000/?config=https%3A%2F%2Fjbrowse.org%2Fgenomes%2FGRCh38%2F1000genomes%2Fconfig_1000genomes.json. 

Could you give me some clue on how to implement this option? Thank you very much!

**Reply by cmdcolin** [reply]
no worries, basically it does not require any special config, it is an alternative build of the codebase that is 'automatically faster'.  the types.frozen stuff is all internal code details and something the config doesnt worry about. so basically, all that is needed to use this branch is to use "jbrowse create --branch frozen_tracks3 newfolder" to create a new instance, and then copy over your config and data files into the newfolder, or run the command "jbrowse upgrade --branch frozen_tracks3" in the same folder as your current instance.

**Reply by cmdcolin** [reply]
(this is the link to the 1000 genomes config just for reference... it is just a normal config.json file and the links i used above were just for demonstrating loading that same config on the current version and the frozen_tracks3 branch https://jbrowse.org/genomes/GRCh38/1000genomes/config_1000genomes.json)

**Comment by crisamor** [comment]
Hi @cmdcolin, sorry for the confusion and for the delay in my response. I'll try to use the frozen_tracks3 branch, let's see if it improves! Thank you very much for your help.
